crates/ty_server/src/document/notebook.rs

```diff
@@ -79,29 +66,38 @@ impl NotebookDocument {
 
     /// Generates a pseudo-representation of a notebook that lacks per-cell metadata and contextual information
     /// but should still work with Ruff's linter.
-    pub fn make_ruff_notebook(&self) -> ruff_notebook::Notebook {
+    pub(crate) fn to_ruff_notebook(&self, index: &Index) -> ruff_notebook::Notebook {
         let cells = self
             .cells
             .iter()
-            .map(|cell| match cell.kind {
-                NotebookCellKind::Code => ruff_notebook::Cell::Code(ruff_notebook::CodeCell {
-                    execution_count: None,
-                    id: None,
-                    metadata: CellMetadata::default(),
-                    outputs: vec![],
-                    source: ruff_notebook::SourceValue::String(
-                        cell.document.contents().to_string(),
-                    ),
-                }),
-                NotebookCellKind::Markup => {
-                    ruff_notebook::Cell::Markdown(ruff_notebook::MarkdownCell {
-                        attachments: None,
+            .map(|cell| {
+                let source = ruff_notebook::SourceValue::String(
+                    index
+                        .document(&DocumentKey::from_url(&cell.url))
+                        .ok()
+                        .and_then(|document| document.as_text())
+                        .map(|text| text.contents().to_string())
+                        .unwrap_or_default(),
```

Comment:
Would it be useful to add a debug log if the index doesn't contain the text document representing the cell?